### ISSUE TITLE
Remove turbo mode from the code base

### ISSUE THREAD

**Issue by jchapma** [389ds/389-ds-base#5852]
**Description.**
Turbo mode is a feature that improves the search performance for a low number of highly active connections. Since its introduction the deployment model has changed and this feature is no longer required. Turbo mode adds some complexity to the code so removal will simplify connection code.

**Solution**
Following investigation into the pros/cons of turbo mode it has been decided that this feature should be removed from the codebase.

**Additional context**
https://bugzilla.redhat.com/show_bug.cgi?id=2225016

**Comment by Firstyear** [comment]
I think we agreed we could remove it if we had epoll rather than select over the fd's since the point of turbo is to avoid the costly call back to select and it setting up the fd arrays. So provided we have epoll working, I'm all for ripping out turbo mode.

**Comment by jchapma** [comment]
HI @Firstyear 
From what I understand there was a performance degradation with the epoll implementation: https://github.com/389ds/389-ds-base/pull/5182#issuecomment-1073948670

**Comment by Firstyear** [comment]
> HI @Firstyear From what I understand there was a performance degradation with the epoll implementation: [#5182 (comment)](https://github.com/389ds/389-ds-base/pull/5182#issuecomment-1073948670)

Yes, but that's because *our* code was creating the performance issues, rather than epoll. We tend to have pretty awful locking through out the code base, so we should be looking to fix epoll here in the long run since it will be a better approach.
